src/main.rs

```diff
@@ -136,6 +137,50 @@ pub fn start(mut os_input: Box<dyn OsApi>, opts: Opt) {
     );
     let maybe_layout = opts.layout.map(Layout::new);
 
+    let send_app_instructions_clone = send_app_instructions.clone();
+
+    #[cfg(not(test))]
+    std::panic::set_hook(Box::new(move |info| {
+        let backtrace = Backtrace::new();
+        let thread = thread::current();
+        let thread = thread.name().unwrap_or("unnamed");
+
+        let msg = match info.payload().downcast_ref::<&'static str>() {
+            Some(s) => *s,
+            None => match info.payload().downcast_ref::<String>() {
+                Some(s) => &**s,
+                None => "Box<Any>",
+            },
+        };
+
+        let backtrace = match info.location() {
+            Some(location) => {
+                format!(
+                    "\nthread '{}' panicked at '{}': {}:{}\n{:?}",
+                    thread,
+                    msg,
+                    location.file(),
+                    location.line(),
+                    backtrace
+                )
+            }
+            None => {
+                format!(
+                    "\nthread '{}' panicked at '{}'\n{:?}",
+                    thread, msg, backtrace
+                )
+            }
+        };
+
+        if send_app_instructions_clone
+            .send(AppInstruction::Error(backtrace.clone()))
+            .is_err()
+        {
+            get_os_input().unset_raw_mode(0);
```

Comment:
I'm not thrilled about this tbh. If I understand correctly, we can't pass os_input here because it doesn't implement `Sync`, just like we had with the app instructions sender, right?

The issue is that while right now calling this factory function doesn't change a lot, in the future we might want to add certain system startup operations there, and then calling it again here might cause issues.

Reply:
Yes the issue is `os_input` does not implement the `Sync` trait. I added this for the case if the `main` thread panics. But on testing this now, I found that it is not working, so the backtrace is getting displayed in a weird manner in this case. How do you propose I proceed here?

Reply:
I think the main thread panicking is a special and rare enough case that we don't need to deal with it right now. It's really okay to just ignore it.